src/features/signUp/SignUpEmail.tsx

```diff
@@ -23,36 +26,82 @@ const SignUpButton = styled.button`
   font-size: 24px;
   font-weight: medium;
   border: none;
-`;
+  cursor: pointer;
 
-const handleSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
-  const formData = new FormData(e.target as HTMLFormElement);
-  const data = Object.fromEntries(formData);
-
-  try {
-    const response = await apiClient.post(API_ENDPOINTS.SIGNUP_EMAIL, {
-      full_name: data.fullName,
-      email: data.email,
-      phone_number: data.phoneNumber,
-      password_hash: data.password,
-    });
-    if (response.status === 200) {
-      // 홈으로 이동
-      console.log('Sign up successful:', response.data);
-    }
-  } catch (error) {
-    console.error('Sign up failed:', error);
+  &:disabled {
+    background-color: #ccc;
+    cursor: not-allowed;
   }
-};
+`;
 
 export default function SignUpEmail() {
+  const [showErrorModal, setShowErrorModal] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+  const [isFullNameValid, setIsFullNameValid] = useState(false);
+  const [isEmailValid, setIsEmailValid] = useState(false);
+  const [isPhoneNumberValid, setIsPhoneNumberValid] = useState(false);
+  const [isPasswordValid, setIsPasswordValid] = useState(false);
+  const [emailError, setEmailError] = useState('');
+
+  const handleSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setIsLoading(true);
+
+    const formData = new FormData(e.target as HTMLFormElement);
+    const data = Object.fromEntries(formData);
```

Comment:
이런 간단한 로직도 외부함수로 분리하면 좋을 것 같아요
```suggestion
    const formData = getFormData(e.target);
    // 컴포넌트 바깥에서 form 데이터 파싱하는 함수
```